pkg/utils/config_checker.go

```diff
@@ -55,9 +56,14 @@ func NewConfigChecker(name string, configfiles ...string) (*configChecker, error
 		name:        name,
 		configfiles: configfiles,
 		checksum:    checksum,
+		reload:      true,
 	}, nil
 }
 
+func (c *configChecker) Reload(reload bool) {
```

Comment:
change name to SetReload or WithReload, and add a comment on what it really means.